test/e2e/common.go

```diff
@@ -143,7 +152,14 @@ func checkNamespaceDeleted(t *testing.T, namespace string, maxRetries int, logge
 	retries := 0
 	for retries < MaxRetries {
 		output, _ := kubectlGetNamespace()
-		if !strings.Contains(output, namespace) {
+
+		// check for namespace deleted
+		if !created && !strings.Contains(output, namespace) {
+			return true
+		}
+
+		// check for namespace created
+		if created && strings.Contains(output, namespace) {
```

Comment:
Can we add at least a minimal sleep (1 or 2s) at least ? Just to avoid to spam the API server. Ideally one would start a watch here for looking for ADD and DELETE events for the namespace, but if you do polling we should be a more patient.

Reply:
Sure, that's reasonable. Will update now.

Reply:
Actually there is a wait between retries in line 169: `time.Sleep(RetrySleepDuration)`

None needed in the `if` statement since that is the positive path :)

Reply:
sorry, that was not clear in the diff.

Reply:
@maximilien however, the retry interval is 30s (which is even longer than the 20s sleep). So when the namespace is not gone after the first request it takes at least 30s before the next try.

I suggest to reduce the retry timeout to 5s if you really aim for a considerable speedup.

Reply:
I refer to this line: https://github.com/knative/client/blob/1406f7de3db975ce2f0c0aae17a60feb1ac46e4e/test/e2e/common.go#L44

Reply:
OK, thx. Pushing update now.